server/athenian/api/openapi/openapi.yaml

```diff
@@ -1435,3 +1513,327 @@ components:
+            login: github.com/gkwillie
+            name: Groundskeeper Willie
+            email: bot@athenian.co
+            timestamp: 2020-02-25T14:27:21Z
+            message: 'Bump version: 0.0.42 → 0.0.43'
+            size_added: 1
+            size_removed: 1
+            files_changed: 1
+      properties:
+        date:
+          description: Commits were pushed beginning with this date. They were not
+            pushed later than +granularity.
+          format: date
+          type: string
+        commits:
+          items:
+            $ref: '#/components/schemas/Commit'
+          type: array
+      required:
+      - commits
+      - date
+      type: object
+    Commit:
+      description: Information about a commit.
+      example:
+        repository: github.com/athenianco/athenian-api
+        hash: 79f0bccb856b35ef08ec93da2ff5585da4d2e50e
+        author:
+          login: github.com/gkwillie
+          name: Groundskeeper Willie
+          email: bot@athenian.co
+          timestamp: 2020-02-25T14:27:21Z
+        committer:
+          login: github.com/gkwillie
+          name: Groundskeeper Willie
+          email: bot@athenian.co
+          timestamp: 2020-02-25T14:27:21Z
+          message: 'Bump version: 0.0.42 → 0.0.43'
+          size_added: 1
+          size_removed: 1
+          files_changed: 1
+      properties:
+        repository:
+          $ref: '#/components/schemas/RepositoryID'
+        hash:
+          description: Git commit hash.
+          type: string
+        author:
+          $ref: '#/components/schemas/CommitSignature'
+        committer:
+          $ref: '#/components/schemas/CommitSignature'
+        message:
+          description: Commit message.
+          type: string
+        size_added:
+          description: Overall number of lines added.
+          type: integer
+        size_removed:
+          description: Overall number of lines removed.
+          type: integer
+        files_changed:
+          description: Number of files changed in this PR.
+          type: integer
+      required:
+      - author
+      - committer
+      - files_changed
+      - hash
+      - message
+      - repository
+      - size_added
+      - size_removed
+      type: object
+    CommitSignature:
+      description: Git commit signature.
+      example:
+        login: github.com/vmarkovtsev
+        name: Vadim Markovtsev
+        email: vadim@athenian.co
+        timestamp: 2020-02-25T14:27:21Z
+      properties:
+        login:
+          $ref: '#/components/schemas/DeveloperID'
+        name:
+          description: Git signature name.
+          type: string
+        email:
+          description: Git signature email.
+          type: string
+        timestamp:
+          description: When the corresponding action happened.
+          format: date-time
+          type: string
+      required:
+      - email
+      - name
+      - timestamp
+      type: object
+    CommitFilter:
```

Comment:
It has some unique props. I will refactor the common filter core later.

Reply:
Could you at least use 
```yml
    MetricsRequest:
      properties:
        for:
            $ref: '#/components/schemas/ForSet'
```
Instead of:
```yml
    MetricsRequest:
      properties:
        in:
          $ref: '#/components/schemas/RepositorySet'
        with_author:
          $ref: '#/components/schemas/DeveloperSet'
        with_committer:
          $ref: '#/components/schemas/DeveloperSet'
```
It would be consistent with the request sent by `/metrics/prs`

Reply:
MetricsRequest is not filtered by commit author and commit committer, so I cannot do it. As I said, I plan to extract the common core for both filter models later, don't worry.

Reply:
That's my point `MetricsRequest` could have been filtered by `commit_author`, `commit_commiter`, `comment_author`, `reviewer`, `merger`, `creator`, but we decided to filter only by `developers` (consider them as "participants"), and let the API do an `OR` with all the different kind of participation in the PR.

In this case, I'm proposing the same. Since there is no necessity on filtering by commit authorship, let's avoid overcomplicating the facade of the API, offering different ways to filter based on the kind of commit authorship.

Reply:
`MetricsRequest` is for pull requests, it is not related to `commit_author` or `commit_commiter`. I will rename it to `PullRequestMetricsRequest` to avoid confusion.

Reply:
we could also filter PRs by kind of participations, PRs are really related to commits and comments.